### PR TITLE
bugfix[DYN-6189]: remove duplicated code block node

### PR DESCRIPTION
**Pull request by Enzo707**
Screen Recording 2023-09-08 at 14 57 54

### Purpose

This PR is for removing the duplicated "code block" node in Library.

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [ ] This PR contains no files larger than 50 MB 

### Release Notes

There was a double indexing statement of the code block node within DynamoModel

### Reviewers

@QilongTang 
@RobertGlobant20 

### FYIs
@avidit

### CODE REVIEW

File: src/DynamoCore/Models/DynamoModel.cs [hunk 1]
```diff
@@ -1594,7 +1594,6 @@ private void InitializeIncludedNodes()
             NodeFactory.AddTypeFactoryAndLoader(outputData.Type);
             NodeFactory.AddAlsoKnownAs(outputData.Type, outputData.AlsoKnownAs);
 
-            SearchModel?.Add(new CodeBlockNodeSearchElement(cbnData, LibraryServices));
```
Comment (QilongTang) [inline src/DynamoCore/Models/DynamoModel.cs:1597]: @Enzo707 after this change, can you test searching for code block node and see if it still works?

Comment (Enzo707) [reply src/DynamoCore/Models/DynamoModel.cs:1597]: @QilongTang if you look at the gif it still appearing in library search and seems to work fine. There's any other place to look at?

Comment (QilongTang) [reply src/DynamoCore/Models/DynamoModel.cs:1597]: yup, can you take a look at the in canvas search as well, just to make sure?

Comment (Enzo707) [reply src/DynamoCore/Models/DynamoModel.cs:1597]: @QilongTang it's failing within canvas search... I'm taking a look

Comment (QilongTang) [reply src/DynamoCore/Models/DynamoModel.cs:1597]: @Enzo707 Any update on this?

Comment (Enzo707) [reply src/DynamoCore/Models/DynamoModel.cs:1597]: @QilongTang sorry for the delay, please take a look.

File: src/DynamoCore/Models/DynamoModel.cs [hunk 2]
```diff
@@ -1613,29 +1615,11 @@ private void InitializeIncludedNodes()
...
-            //Adding this nodes are breaking the tests (due that we have two input and two output nodes):
-            //WhenHomeWorkspaceIsFocusedInputAndOutputNodesAreMissingFromSearch
-            //WhenStartingDynamoInputAndOutputNodesAreNolongerMissingFromSearch
-            // New index process from Lucene, adding missing nodes: Code Block, Input and Output
-            var ele = AddNodeTypeToSearch(outputData);      
-            if (ele != null)
-            {
-                AddNodeTypeToSearchIndex(ele, iDoc);
-            }
+            AddNodeTypeToSearchIndex(symbolSearchElement, iDoc);
```
Comment (QilongTang) [inline src/DynamoCore/Models/DynamoModel.cs:1618]: @RobertGlobant20 Didn't we add the check intentionally? Should we merge this only after we have the corresponding unit test added?

Comment (RobertGlobant20) [reply src/DynamoCore/Models/DynamoModel.cs:1618]: If you are refering to the check:
`if (ele != null)`
I think there is no need to check it due that we are not using the AddNodeTypeToSearch(outputData); method  anymore (that could return null) and if you see the code some lines above you will see that those instances are created (see screenshot below).

The unit test only validates that the Input, Output and Code Block nodes are found with the Lucene Search results then it won't catch if they are duplicated visually in Library (As far as I remember Lucene Search doesn't return duplicated items), then I think this change can be merged

image

**Review by QilongTang — APPROVED** [review]

**Pull request merged**